content/guides/general/inline-blocks/inline-form-fields.md

```diff
@@ -0,0 +1,150 @@
+    <div className="home">
+      <Hero data={pageData.data} />
+    </div>
+  )
+}
+```
+
+First, import `useForm` from 'tinacms'. Then create a formConfig object where you set the `id` and `initialValues` to connect the form with the data.
+
+Call the `useForm` hook and pass the config object. The return values will be the data and the form itself. We will use the `form` next for `InlineForm`.
+
+## Create an _Inline Form_
+
+Now let's add an [_Inline Form_](https://tinacms.org/docs/inline-editing) to the home page. First, install the `react-tinacms-inline` package.
+
+```bash
+yarn add react-tinacms-inline
+```
+
+Wrap `InlineForm` around the `Hero` component, and pass the `form`.
+
+**Home.js**
+
+```jsx
+import React from 'react'
+import { useForm, usePlugin } from 'tinacms'
+// 1. Import `InlineForm`
+import { InlineForm } from 'react-tinacms-inline'
+import { Hero } from './components/Hero'
+import data from './data/data.json'
+
+export default function Home() {
+  const formConfig = {
+    id: './data/data.json',
+    initialValues: {
+      data,
+    },
+    onSubmit() {
+      alert('Saved!')
+    },
+  }
+
+  const [pageData, form] = useForm(formConfig)
+  usePlugin(form)
+
+  // 2. Wrap `InlineForm` around `Hero`, pass the form
+  return (
+    <div className="home">
+      <InlineForm form={form}>
+        <Hero data={pageData.data} />
+      </InlineForm>
+    </div>
+  )
+}
+```
+
+If you restart the dev server, things won't look different. But changes have happened under the hood — the data being rendered by `Hero` is now _hooked up_ to an _Inline Form_ with the CMS.
+
+At this point, we've **turned the entire home page into a form**. So any _Inline Fields_ we add to this on-page form will be editable directly on the page.
+
+## Add _Inline Fields_
+
+Now that we have the _Inline Form_ set up, we can add fields to it.
+
+_Inline Fields_ are essentially various types of inputs, with stripped-down styles so they blend in with the page. You can [create your own manually](https://tinacms.org/docs/inline-editing#adding-inline-editing-with-inlineform), or use the [pre-configured fields](https://tinacms.org/docs/inline-editing#using-pre-configured-inline-fields) provided by `react-tinacms-inline`. For the demo, we will only be using the pre-configured fields.
+
+Head to the `Hero` component. Let's turn the headline and subtext into _Inline Textareas_:
+
+**components/Hero.js**
+
+```diff
+import React from 'react'
+// 1. Import `InlineTextarea`
++ import { InlineTextarea } from 'react-tinacms-inline'
+import '../styles/hero.css'
+
+- export function Hero({data}) {
++ export function Hero() {
+
+  // 2. Replace `data` with Inline Fields
+  return (
+    <div className="hero">
+      <div className="wrapper wrapper--narrow">
+-       <h1>{data.headline}</h1>
++       <h1>
++        <InlineTextarea name="headline" />
++       </h1>
+-       <p>{data.subtext}</p>
++       <p>
++         <InlineTextarea name="subtext" />
++       </p>
```

Comment:
```suggestion
+        <InlineTextarea name="data.headline" />
+       </h1>
-       <p>{data.subtext}</p>
+       <p>
+         <InlineTextarea name="data.subtext" />
+       </p>
```

This is the only way this works for me. I _don't think_ I've missed a step or anything.

Reply:
yeah the example is messed up, we're fixing it 👌